core/trino-main/src/main/java/io/trino/operator/DeduplicatingDirectExchangeBuffer.java

```diff
@@ -645,14 +660,21 @@ public void close()
             pageBufferRetainedSizeInBytes = 0;
             bufferedPageCount = 0;
             writeBuffer = null;
-            try (Closer closer = Closer.create()) {
-                closer.register(exchange);
-                if (exchangeSink != null) {
-                    closer.register(exchangeSink::abort);
+
```

Comment:
can it still be null? You removed null assignment in https://github.com/trinodb/trino/pull/10699/commits/8415672457b4ac9dd963749b15b08cae8175b376#diff-b366169ac97dd5dbdbbe1691dc5054236f8e24f451abb8a298d26e4b811d213cL604